### DISCUSSION TITLE
How to get the e-tag during a `get_opts` operation?

### DISCUSSION THREAD

**Discussion by winding-lines** [apache/arrow-rs#4495] [Q&A] [answered]
Here is the flow I am trying to implement:

1. when I download an object the first time I would like to save its e-tag and content
2. on subsequent downloads I would like to use `if_none_match` 
 a. if the tag has changed save the new content and e-tag
 b. otherwise the api will fail and I can use the local data

I don't seem to have a way to get the e-tag from a `get` or `get_opts` API call. The docs and tests I found indicate that I need to do a `head` to get the metadata for a given object and get the e-tag from the metadata. But this increases the number of API calls, if `get` had a way to return the e-tag I could implement the logic above just will calls to `get`.

Am I missing something?

Thanks for your advice.

**Comment by tustvold** [comment]
Unfortunately there isn't a way to do this currently, but perhaps this is something that could be added as part of https://github.com/apache/arrow-rs/issues/4352

**Comment by tustvold** [comment, marked as answer]
https://github.com/apache/arrow-rs/pull/4677 proposes an API change that will provide access to the etag on get
